modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+		return TRUE */ // Disabled, uncomment if we want this.
+
+/mob/living/carbon/death(gibbed) // On death, we clear the indiciators
+	..() // Call the TG death. Do not . = ..()!
+	for(var/iteration in status_indicators) // When we die, clear the indicators.
+		remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+/* /datum/status_effect/incapacitating/on_apply()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 100 MILLISECONDS)
+/datum/status_effect/incapacitating/on_remove()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 1000 MILLISECONDS) */
+
+/* /mob/living/carbon/apply_effect()
+	. = ..()
+	status_sanity() */
+
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CARBON_HEALTH_UPDATE, .proc/status_sanity)
+
+/* /mob/living/updatehealth()
+	. = ..()
+	status_sanity() */
+/mob/living/proc/status_sanity()
+	SIGNAL_HANDLER
+	if(stat == DEAD)
+		for(var/iteration in status_indicators) // When we die, clear the indicators.
+			remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+	is_weakened() ? add_status_indicator("weakened") : remove_status_indicator("weakened")
+	is_paralyzed() ? add_status_indicator("paralysis") : remove_status_indicator("paralysis")
+	is_stunned() ? add_status_indicator("stunned") : remove_status_indicator("stunned")
+	is_unconcious() ? add_status_indicator("sleeping") : remove_status_indicator("sleeping")
+	is_confused() ? add_status_indicator("confused") : remove_status_indicator("confused")
+//	is_blind_status() ? add_status_indicator("blinded") : remove_status_indicator("blinded") // Disabled, keeping for IF we decide we want this.
+//	is_critical() ? add_status_indicator("critical") : remove_status_indicator("critical") // Also disabled. Crosses medhud territory, despite being a shift click examine thing.
+
+/mob/living/proc/add_status_indicator(image/thing)
+	if(get_status_indicator(thing)) // No duplicates, please.
+		return
+	if(!istype(thing, /image))
+		thing = image(icon = 'modular_skyrat/master_files/icons/mob/status_indicators.dmi', loc = src, icon_state = thing)
+
+	LAZYADD(status_indicators, thing)
+	handle_status_indicators()
+
+// Similar to above but removes it instead, and nulls the list if it becomes empty as a result.
+/mob/living/proc/remove_status_indicator(image/thing)
+	thing = get_status_indicator(thing)
+
+	cut_overlay(thing)
+	LAZYREMOVE(status_indicators, thing)
+	handle_status_indicators()
+
+/mob/living/proc/get_status_indicator(image/thing)
+	if(!istype(thing, /image))
+		for(var/image/I in status_indicators)
+			if(I.icon_state == thing)
+				return I
+	return LAZYACCESS(status_indicators, LAZYFIND(status_indicators, thing))
```

Comment:
Ditto, document this and give `thing` a better name, while also making it not a `/image`. You could also even go a step further and add `RETURN_TYPE(/image)`, which I'll show you how to do, and that allows you to access variables on the return value of that proc without having to store it in a variable that's typed to `/image`.

Here's a quick suggestion to clear that single-letter variable because that's also pretty bad. (We use `as anything` because we know there's only going to be `/image`-type elements in that list (if not, that's something you should fix).

Mind you, I didn't remove the fact that they can be images, mostly because I'm not certain of how it could break your code.
```suggestion
/mob/living/proc/get_status_indicator(image/thing)
	RETURN_TYPE(/image)
	if(!istype(thing, /image))
		for(var/image/indicator as anything in status_indicators)
			if(indicator.icon_state == thing)
				return indicator
	return LAZYACCESS(status_indicators, LAZYFIND(status_indicators, thing))
```